src/flags.jl

```diff
@@ -0,0 +1,226 @@
+        (:READ1,         UInt16(0x040), "this is read1"                                                             ),
+        (:READ2,         UInt16(0x080), "this is read2"                                                             ),
+        (:SECONDARY,     UInt16(0x100), "not primary alignment"                                                     ),
+        (:QCFAIL,        UInt16(0x200), "QC failure"                                                                ),
+        (:DUP,           UInt16(0x400), "optical or PCR duplicate"                                                  ),
+        (:SUPPLEMENTARY, UInt16(0x800), "supplementary alignment"                                                   ),]
+    @assert bits isa UInt16 "The bits must be of type UInt16."
+    sym = Symbol("FLAG_", name)
+    docstring = """    $sym
+    SAM/BAM flag: $doc
+
+    See also: [`flag`](@ref)
+    """
+    @eval begin
+        @doc $(docstring) const $(sym) = $(bits)
+    end
+end
+
+"""
+    ispaired(record::XAMRecord)::Bool
+
+Query whether the `record`'s template has multiple segments in sequencing.
+"""
+function ispaired(record::XAMRecord)::Bool
+    return flag(record) & FLAG_PAIRED == FLAG_PAIRED
+end
+
+"""
+    isproperpair(record::XAMRecord)::Bool
+
+Query whether each segment of the `record`'s template properly aligned according to the aligner.
+"""
+function isproperpair(record::XAMRecord)::Bool
+    return flag(record) & PROPER_PAIR == PROPER_PAIR
+end
+
+"""
+    isunmapped(record::XAMRecord)::Bool
+
+Query whether the `record` is unmapped.
+"""
+function isunmapped(record::XAMRecord)::Bool
+    return flag(record) & FLAG_UNMAP == FLAG_UNMAP
+end
+
+"""
+    ismapped(record::XAMRecord)::Bool
+
+Query whether the `record` is mapped.
+"""
+function ismapped(record::XAMRecord)::Bool
+    # return flag(record) & FLAG_UNMAP == 0
+    return isfilled(record) && (flag(record) & FLAG_UNMAP == 0)
+end
+
+"""
+    ismateunmapped(record::XAMRecord)::Bool
+
+Query whether the `record`'s mate is unmapped.
+"""
+function ismateunmapped(record::XAMRecord)::Bool
+    return flag(record) & FLAG_MUNMAP == FLAG_MUNMAP
+end
+
+"""
+    isnextmapped(record::XAMRecord)::Bool
+
+Test if the mate/next read of `record` is mapped.
+"""
+function isnextmapped(record::XAMRecord)::Bool
+    return flag(record) & FLAG_MUNMAP == 0
+end
+
+"""
+    isreverse(record::XAMRecord)::Bool
+
+Query whether the `record` is mapped to the reverse strand.
+"""
+function isreverse(record::XAMRecord)::Bool
+    return flag(record) & FLAG_REVERSE == FLAG_REVERSE
+end
+
+"""
+    isforward(record::XAMRecord)::Bool
+
+Query whether the `record` is mapped to the forward strand.
+"""
+function isforward(record::XAMRecord)::Bool
+    return flag(record) & FLAG_REVERSE == 0
+end
+
+"""
+    ispositivestrand(record::XAMRecord)::Bool
+
+Query whether `record` is aligned to the positive strand.
+"""
+function ispositivestrand(record::XAMRecord)::Bool
+    return isforward(record)
+end
```

Comment:
Are these aliases necessary? It might be less confusing to pick one term (reverse or negative) and stick to it

Reply:
It would be good to ensure the core functions match the SAM spec, but aliases could help newcomers who may be building an understanding of different terms.

Reply:
The `isforward` method was renamed as `isforwardstrand` and now interprets `isreversecomplemented`.
I've added a `isreversestrand` method, that also interprets `isreversecomplemented`. The `ispositivestrand` method aliases `isforwardstrand`. Likewise `isnegativestrand` aliases `isreversestrand`.